core/src/main/java/org/fao/geonet/kernel/Thesaurus.java

```diff
@@ -693,6 +693,47 @@ public void addTitleElement(String thesaurusTitle) throws IOException, AccessDen
         repository.addGraph(myGraph);
     }
 
+    //    <rdf:RDF>
+    //    <skos:ConceptScheme rdf:about="http://www.thesaurus.gc.ca/#CoreSubjectThesaurus">
+    //          <dc:title>Main GN Title</dc:title>
+    //          <dc:title xml:lang="en">English Version (en)</dc:title>
+    //          <dc:title xml:lang="fr">French Version (fr)</dc:title>
+    //          <dc:description></dc:description>
+    //     </skos:ConceptScheme>
+    //
+    // This will setup;
+    //  {
+    //      "en": "English Version (en)",
+    //      "fr": "French Version (fr)"
+    //      "eng": "English Version (en)",
+    //      "fre": "French Version (fr)"
+    //  }
+    private void retrieveMultiLingualTitles(Element thesaurusEl) {
+        List<Namespace> theNSs = new ArrayList<Namespace>();
+        theNSs.add(Namespace.getNamespace("rdf", "http://www.w3.org/1999/02/22-rdf-syntax-ns#"));
+        theNSs.add(Namespace.getNamespace("skos", "http://www.w3.org/2004/02/skos/core#"));
+        theNSs.add(Namespace.getNamespace("dc", "http://purl.org/dc/elements/1.1/"));
+        theNSs.add(Namespace.getNamespace("dcterms", "http://purl.org/dc/terms/"));
+
+        Namespace xmlNS = Namespace.getNamespace("xml","http://www.w3.org/XML/1998/namespace");
+
+        try {
+            List<Element> multiLingualTitles = (List<Element>) Xml.selectNodes(thesaurusEl,
+                "skos:ConceptScheme/dc:title[@xml:lang]|skos:ConceptScheme/dcterms:title[@xml:lang]", theNSs);
+            multilingualTitles.clear();
+            for (Element el: multiLingualTitles) {
+                    //use both forms of the lang name
```

Comment:
Are both required? Can't be send with the iso code used by other thesaurus services and use in the UI the services to convert the codes?

Reply:
I put this in for some consistency - dealing with any form of language codes can be difficult (i.e. fr/fre/fra).  This makes things easier down the line.

Reply:
I added a bit of code so it wouldn't overwrite explicitly set languages using this 2/3 letter conversion.

Reply:
Would it not be better to create 2 functions? - one to get 2 char lang code and values like the following
```
    {
       "en": "English Version (en)",
       "fr": "French Version (fr)"
    }

```
and one to get 3 char lang codes and values like the following

 ```
    {
       "eng": "English Version (en)",
       "fre": "French Version (fr)"
    }

```
  With 2 function, if in the future you want to know where 2 char codes are used vs 3 char code it will be easier to identify usage. This can probably be done by making the main function private and creating 2 public functions that could just subset the existing results.

Reply:
I can just put in the values that were explicitly specified.  However, this is going to make the XSLT code much more difficult.

Reply:
Ok, I pushed a commit to handle this.  There's will now only be "en" and "fr" in the Thesaurus object